### PR TITLE
Add SOS support

### PR DESCRIPTION
**Pull request by pangea-andrest**

### CODE REVIEW

File: pangea-sdk/v3/service/store/integration_test.go [hunk 1]
```diff
@@ -0,0 +1,413 @@
...
+	assert.NotNil(t, respPut2)
+	assert.NotNil(t, respPut2.Result)
+	assert.Equal(t, folderID, respPut2.Result.Object.ParentID)
+	assert.Equal(t, store.Metadata(METADATA), respPut2.Result.Object.Metadata)
+	assert.Equal(t, store.Tags(TAGS), respPut2.Result.Object.Tags)
+	assert.Empty(t, respPut2.Result.Object.MD5)
+	assert.Empty(t, respPut2.Result.Object.SHA512)
+	assert.NotEmpty(t, respPut2.Result.Object.SHA256)
+
+	// Update file. full metadata and tags
```
Comment (kenany) [inline pangea-sdk/v3/service/store/integration_test.go:274]: (nitpick) maybe?:
```suggestion
	// Update file with full metadata and tags.
```
edit: repeats elsewhere below as well

**Review by kenany — APPROVED** [review]

File: pangea-sdk/v3/service/vault/integration_test.go [hunk 2]
```diff
@@ -24,7 +24,7 @@ const (
 )
 
 var timeNow = time.Now()
-var timeStr = timeNow.Format("yyyyMMdd_HHmmss")
+var timeStr = timeNow.Format("20060102_150405")
```
[the remark this answers is not in the document]
Comment (pangea-andrest) [reply pangea-sdk/v3/service/vault/integration_test.go:27]: It's used to create unique names for keys

[the remark this answers is not in the document]
Comment (kenany) [reply pangea-sdk/v3/service/vault/integration_test.go:27]: I mean what's the change for?

[the remark this answers is not in the document]
Comment (pangea-andrest) [reply pangea-sdk/v3/service/vault/integration_test.go:27]: https://www.geeksforgeeks.org/time-formatting-in-golang/ to use proper formatting. If you mean if it's needed or not in the over all test, we can review in detail if we want to keep it or not.

File: pangea-sdk/v3/service/store/integration_test.go (hunk 1 above)
Comment (pangea-andrest) [reply pangea-sdk/v3/service/store/integration_test.go]: fixed

File: pangea-sdk/v3/service/vault/integration_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (kenany) [reply pangea-sdk/v3/service/vault/integration_test.go:27]: So `"yyyyMMdd_HHmmss"` was improper? Got it.

**Review by kenany — APPROVED** [review]

File: pangea-sdk/v3/service/store/api.go [hunk 3]
```diff
@@ -0,0 +1,614 @@
...
+	Tags           Tags     `json:"tags,omitempty"`
+	ParentID       string   `json:"parent_id,omitempty"`
+	UpdatedAt      string   `json:"updated_at,omitempty"`
+}
+
+type UpdateResult struct {
+	Object ItemData `json:"object"`
+}
+
+func (e *store) Update(ctx context.Context, input *UpdateRequest) (*pangea.PangeaResponse[UpdateResult], error) {
```
Comment (r3dcrosse) [inline pangea-sdk/v3/service/store/api.go:212]: I think Update is missing a parameter for the updated file?

Screenshot 2024-01-09 at 1 35 48 PM

**Review by kenany — APPROVED** [review]

File: pangea-sdk/v3/service/store/api.go (hunk 3 above)
Comment (pangea-andrest) [reply pangea-sdk/v3/service/store/api.go:307]: I think it's not supported yet, checking with Jason...

Comment (pangea-andrest) [reply pangea-sdk/v3/service/store/api.go:307]: No, it's not.

Comment (r3dcrosse) [reply pangea-sdk/v3/service/store/api.go:307]: Thanks for checking. I'll go ahead and push a commit to update the doc string here.

**Review by kenany — DISMISSED** [review]

File: examples/README.md [hunk 4]
```diff
@@ -4,7 +4,7 @@ This is a quick example about how you use Pangea Go SDK, set up and run it.
 
 ## Setup
 
-Set up environment variables ([Instructions](https://pangea.cloud/docs/getting-started/integrate/#set-environment-variables)) `PANGEA_AUDIT_TOKEN` and `PANGEA_DOMAIN` with your project token configured on Pangea User Console (token should have access to Audit service [Instructions](https://pangea.cloud/docs/getting-started/configure-services/#configure-a-pangea-service)) and with your pangea domain.
+Set up environment variables ([Instructions](https://pangea.cloud/docs/getting-started/integrate/#set-environment-variables)) `PANGEA_<service>_TOKEN` and `PANGEA_DOMAIN` with your project token configured on Pangea User Console (token should have access to <service> [Instructions](https://pangea.cloud/docs/getting-started/configure-services/#configure-a-pangea-service)) and with your pangea domain.
```
Comment (kenany) [inline examples/README.md:7]: Note that the second `<service>` does not render on GitHub.
image

Comment (pangea-andrest) [reply examples/README.md]: Updated merging main

**Review by kenany — APPROVED** [review]

**Review by kenany — DISMISSED** [review]

**Review by kenany — DISMISSED** [review]

**Review by kenany — DISMISSED** [review]

**Review by kenany — DISMISSED** [review]

**Review by kenany — DISMISSED** [review]

**Review by kenany — APPROVED** [review]

**Review by kenany — DISMISSED** [review]

**Review by kenany — APPROVED** [review]

**Pull request merged**